app/models/lookup_keys/lookup_key.rb

```diff
@@ -76,6 +76,10 @@ def self.find_by_name(str)
     nil
   end
 
+  def reject_invalid_lookup_values(attributes)
```

Comment:
Removing the `reject_if` completely is what caused the integration test to fail.
When saving through the UI there wasn't a problem but in the integration test the fact that there is a hidden matcher (that is shown once you click `+ Add Matcher` cause the form to refuse to be saved because of the validation added on lookup_value for empty values.
The hidden matcher has `:match => "", :value => ""` and tries to update on save and of course fails.
Adding `reject_if` to reject all lookup_values with empty matcher fixes the test.
This is also good to have since the validation on lookup_value is only for `PuppetclassLookupKey` and we still want to reject any type of lookup_key that doesn't contain a matcher.